src/wwx/2016/Justin Donaldson.md

```diff
@@ -0,0 +1,156 @@
+[_template]: ../../templates/interview.html
+
+Tell us about yourself?
+How did you get started in programming? and/or How did you get started in your specific field?
+I grew up in a small sized town in the midwest of the US.  Computers were my way of connecting to like minded people, and learning new things in a relatively isolated hometown.  I've also done a lot of graduate work, and was one of the first PhDs from Indiana University's new Informatics School (Computer Science, BioInformatics, HCI, Cybersecurity, etc.).   
+
+What is your job?
+My official title is Lead Data Scientist or Machine Learning engineer, but I still specialize in being able to make engineering changes necessary for Machine Learning to be possible on large systems.  So, it's a combination of coding and computational statistics.
+ 
+Who do you work for?
+Salesforce.com 
+Do you use Haxe at work?
+No, or at least on very rare occasions :(
+If you do, how?
+If not, are there any areas you want and/or can see Haxe fitting in perfectly?
+
+There are plenty of in-house frameworks where we're trying to formalize javascript to make it amenable for giant multi-team efforts on client side apps.  Haxe is a good fit in many ways for that, but is an unlikely choice for a giant enterprise software company that intends to stick with what it perceives as a safe choice.  It doesn't stop me from trying to pitch Haxe there though :)
+ 
+Out of the various Haxe IDE's available, which one(s) do you use?
+
+I use my own vaxe plugin for vim exclusively 
+If you use more than one, why?
+What other software do you find vital while working with Haxe?
+
+Besides standard browsers, I rarely use much else.  In some rare cases I might use wireshark or some special network tools to develop wire protocols for databases, etc.
+Do you integrate any cloud based services while working with Haxe?
+I use Github and Travis pretty heavily.  They've improved the quality of my code and workflow tremendously.
+ 
+And what problem do they solve for you?
+Github is the social life blood for an open source project.  Previously only large scale projects could expect to see contributions from random people.  Now it feels like anyone can contribute anywhere, which is incredible.  Having a well organized github project is a great sign of its quality.  Having a travis test suite for the library is another great sign, and it makes a lot of cross-platform dev transparent and sane.
+ 
+What hardware you do you use?
+
+Mostly mac laptops.  I have several.
+ 
+What problem does Haxe solve for you?
+
+Haxe helps me compose solutions to problems consistently, using a range of OOP, functional, and dynamic approaches.  The compiler takes over a wide range of type checking and internal consistency checks without requiring a burden of verbose code.  It's wonderful to have the extra bandwidth back for my brain. 
```

Comment:
I haven't put much thought into new core features yet.  Most of my wish list right now includes better tooling support, detecting unused imports and the like.  Some of that might involve compiler support though, since really only the compiler knows exactly where a given type came from, and where it was used in the current file.

I see multi returns as being a special case for a few targets that support them (e.g. lua and python).  For that reason, I don't think multi returns will become a core feature.  For targets that didn't support multi-returns natively, there would be no performance gain, and I don't believe multi-returns would add clarity or convenience to the type information for a given function.  Multi returns are important for externs in a few cases though, so I think it's worth tackling as an extern-only feature.